### DISCUSSION TITLE
Saving profile data on signup

### DISCUSSION THREAD

**Discussion by oProtica** [supabase/supabase#2292] [Questions]
So I am aware that you can store profile data like usernames and avatar urls in a separate public table and I know how to set up database functions to automatically create a row in the public profiles table whenever a row is created in the auth users table and pass the user id while doing that. The problem with this is that users can't signup with a unique username like many other sites. The alternatives are just leaving the username column null and allowing the user to configure it after account creation. The only way I could think to combat this is by querying the profiles table to check if a username is available, then creating the account using which will automatically make an row in the profiles table with no username, and finally modifying the username in the profiles table. There are still problems with this approach as the username won't be modifiable until the user has verified their email unless you disable email confirmations. It would be great if there was built in support for usernames as I am sure many apps would use it but this isn't a feature request, I am just looking for any other ideas on ways to achieve what I am trying to do.

**Comment by CaseyG5** [comment]
Sounds like you answered your own question.  I think most if not all systems requiring a unique username will have to make a query to ensure the one chosen is available.  In any event, I agree that asking for some sort of name, whether full name or nickname or whatever, is a VERY common use case.  Personally, I decided to turn off email confirmations to simplify testing.
